test/index.js

```diff
@@ -21,334 +21,297 @@ const expect = Code.expect;
+            await client.drop({});
+        }
+        catch (err) {
+            expect(err instanceof Error).to.equal(true);
+        }
     });
 
-    it('returns error on set when using invalid key', (done) => {
+    it('returns error on set when using invalid key', async () => {
 
         const client = new Catbox.Client(Mongo);
-        client.start((err) => {
-
-            expect(err).to.not.exist();
-            client.set({}, {}, 1000, (err) => {
-
-                expect(err instanceof Error).to.equal(true);
-                done();
-            });
-        });
+        await client.start();
+
+        try {
+            await client.set({}, {}, 1000);
+        }
+        catch (err) {
+            expect(err instanceof Error).to.equal(true);
+        }
     });
 
-    it('ignores set when using non-positive ttl value', (done) => {
+    it('ignores set when using non-positive ttl value', async () => {
 
         const client = new Catbox.Client(Mongo);
-        client.start((err) => {
-
-            expect(err).to.not.exist();
-            const key = { id: 'x', segment: 'test' };
-            client.set(key, 'y', 0, (err) => {
+        await client.start();
 
-                expect(err).to.not.exist();
-                done();
-            });
-        });
+        const key = { id: 'x', segment: 'test' };
+        await client.set(key, 'y', 0);
     });
 
-    it('returns error on drop when using null key', (done) => {
+    it('returns error on drop when using null key', async () => {
 
         const client = new Catbox.Client(Mongo);
-        client.start((err) => {
-
-            expect(err).to.not.exist();
-            client.drop(null, (err) => {
-
-                expect(err instanceof Error).to.equal(true);
-                done();
-            });
-        });
+        await client.start();
+        try {
+            await client.drop(null);
+        }
+        catch (err) {
+            expect(err).to.exist();
+        }
     });
 
-    it('returns error on get when stopped', (done) => {
+    it('returns error on get when stopped', async () => {
 
         const client = new Catbox.Client(Mongo);
         client.stop();
         const key = { id: 'x', segment: 'test' };
-        client.connection.get(key, (err, result) => {
-
+        try {
+            await client.get(key);
+        }
+        catch (err) {
             expect(err).to.exist();
-            expect(result).to.not.exist();
-            done();
-        });
+        }
     });
 
-    it('returns error on set when stopped', (done) => {
+    it('returns error on set when stopped', async () => {
 
         const client = new Catbox.Client(Mongo);
         client.stop();
         const key = { id: 'x', segment: 'test' };
-        client.connection.set(key, 'y', 1, (err) => {
 
+        try {
+            await client.set(key, 'y', 1);
+        }
+        catch (err) {
             expect(err).to.exist();
-            done();
-        });
+        }
```

Comment:
If client.set is successful the test will pass. Maybe better pattern is to use expects().to.reject() for these scenarios. Like this:
https://github.com/hapijs/catbox-memory/commit/b89f356972116cd18f149e90e0a6cc08e036da22#diff-910eb6f57886ca16c136101fb1699231R165

Reply:
You have this pattern on several places, please review also these other cases.

Reply:
Thank you for pointing me to `expect().to.reject()`. Didn’t know it before 👍